ui: watch SSE label selector has leading comma — floods console with Bad Request errors

**Expected:**
The agent (and other resource) watch streams should connect cleanly and only reconnect after a genuine disconnect, using a reasonable backoff.

**Actual:**
In `src/app/api/watch/agents/route.ts` the label selector is built with a leading comma:

\`\`\`typescript
let labelSelector = \`\`
if (clusterName) {
  labelSelector += \`,langop.io/cluster=\${clusterName}\`  // leading comma → ",langop.io/cluster=test-cluster"
}
\`\`\`

K8s rejects this as `400 Bad Request`. The error handler retries after only 100ms (line 108), and simultaneously the client-side EventSource reconnects on each error event — together producing a tight loop that floods the browser console with 90+ errors per second:

```
[ERROR] Watch error event: {type: error, message: Bad Request, ...}
[ERROR] EventSource error (readyState: OPEN): {url: /api/watch/agents?cluster=test-cluster, ...}
```

This also affects `/api/watch/tools`, `/api/watch/models`, `/api/watch/personas` which have the same pattern.

**Acceptance Criteria:**
- [ ] Label selector is built without a leading comma (e.g., initialize to `undefined` and only set when `clusterName` is present, or trim the leading comma)
- [ ] Server-side watch retry uses exponential backoff (minimum 1s, not 100ms) for persistent errors
- [ ] Browser console is clean when navigating to any cluster sub-page

--- a/src/app/api/watch/agents/route.ts
+++ b/src/app/api/watch/agents/route.ts
@@ -20,6 +20,7 @@ export async function GET(request: NextRequest) {
     let watchCleanup: (() => void) | null = null
     let retryTimeout: NodeJS.Timeout | null = null
     let lastResourceVersion: string | undefined = undefined
+    let retryCount = 0
 
     // Create safe SSE stream with automatic lifecycle management
     const { stream, sendEvent, isActive } = createSSEWatchStream(request, {
@@ -50,10 +51,7 @@ export async function GET(request: NextRequest) {
 
       try {
         // Build label selector
-        let labelSelector = ``
-        if (clusterName) {
-          labelSelector += `,langop.io/cluster=${clusterName}`
-        }
+        const labelSelector = clusterName ? `langop.io/cluster=${clusterName}` : undefined
 
         watchCleanup = await watchService.watchLanguageAgents(
           {
@@ -104,11 +102,14 @@ export async function GET(request: NextRequest) {
 
             // Retry only if client is still connected
             if (!request.signal.aborted && isActive()) {
-              console.log('🔄 Reconnecting agent watch immediately...')
-              retryTimeout = setTimeout(startWatch, 100) // 100ms delay to prevent tight loop
+              const delay = Math.min(1000 * Math.pow(2, retryCount), 30000)
+              retryCount++
+              console.log(`🔄 Reconnecting agent watch in ${delay}ms...`)
+              retryTimeout = setTimeout(startWatch, delay)
             }
           }
         )
+        retryCount = 0
       } catch (error) {
         console.error('Failed to start agent watch:', error)
 

--- a/src/app/api/watch/models/route.ts
+++ b/src/app/api/watch/models/route.ts
@@ -20,6 +20,7 @@ export async function GET(request: NextRequest) {
     let watchCleanup: (() => void) | null = null
     let retryTimeout: NodeJS.Timeout | null = null
     let lastResourceVersion: string | undefined = undefined
+    let retryCount = 0
 
     // Create safe SSE stream with automatic lifecycle management
     const { stream, sendEvent, isActive } = createSSEWatchStream(request, {
@@ -50,10 +51,7 @@ export async function GET(request: NextRequest) {
 
       try {
         // Build label selector
-        let labelSelector = ``
-        if (clusterName) {
-          labelSelector += `,langop.io/cluster=${clusterName}`
-        }
+        const labelSelector = clusterName ? `langop.io/cluster=${clusterName}` : undefined
 
         watchCleanup = await watchService.watchLanguageModels(
           {
@@ -104,11 +102,14 @@ export async function GET(request: NextRequest) {
 
             // Retry only if client is still connected
             if (!request.signal.aborted && isActive()) {
-              console.log('🔄 Reconnecting model watch immediately...')
-              retryTimeout = setTimeout(startWatch, 100) // 100ms delay to prevent tight loop
+              const delay = Math.min(1000 * Math.pow(2, retryCount), 30000)
+              retryCount++
+              console.log(`🔄 Reconnecting model watch in ${delay}ms...`)
+              retryTimeout = setTimeout(startWatch, delay)
             }
           }
         )
+        retryCount = 0
       } catch (error) {
         console.error('Failed to start model watch:', error)
 

--- a/src/app/api/watch/personas/route.ts
+++ b/src/app/api/watch/personas/route.ts
@@ -20,6 +20,7 @@ export async function GET(request: NextRequest) {
     let watchCleanup: (() => void) | null = null
     let retryTimeout: NodeJS.Timeout | null = null
     let lastResourceVersion: string | undefined = undefined
+    let retryCount = 0
 
     // Create safe SSE stream with automatic lifecycle management
     const { stream, sendEvent, isActive } = createSSEWatchStream(request, {
@@ -50,10 +51,7 @@ export async function GET(request: NextRequest) {
 
       try {
         // Build label selector
-        let labelSelector = ``
-        if (clusterName) {
-          labelSelector += `,langop.io/cluster=${clusterName}`
-        }
+        const labelSelector = clusterName ? `langop.io/cluster=${clusterName}` : undefined
 
         watchCleanup = await watchService.watchLanguagePersonas(
           {
@@ -104,11 +102,14 @@ export async function GET(request: NextRequest) {
 
             // Retry only if client is still connected
             if (!request.signal.aborted && isActive()) {
-              console.log('🔄 Reconnecting persona watch immediately...')
-              retryTimeout = setTimeout(startWatch, 100) // 100ms delay to prevent tight loop
+              const delay = Math.min(1000 * Math.pow(2, retryCount), 30000)
+              retryCount++
+              console.log(`🔄 Reconnecting persona watch in ${delay}ms...`)
+              retryTimeout = setTimeout(startWatch, delay)
             }
           }
         )
+        retryCount = 0
       } catch (error) {
         console.error('Failed to start persona watch:', error)
 

--- a/src/app/api/watch/tools/route.ts
+++ b/src/app/api/watch/tools/route.ts
@@ -20,6 +20,7 @@ export async function GET(request: NextRequest) {
     let watchCleanup: (() => void) | null = null
     let retryTimeout: NodeJS.Timeout | null = null
     let lastResourceVersion: string | undefined = undefined
+    let retryCount = 0
 
     // Create safe SSE stream with automatic lifecycle management
     const { stream, sendEvent, isActive } = createSSEWatchStream(request, {
@@ -50,10 +51,7 @@ export async function GET(request: NextRequest) {
 
       try {
         // Build label selector
-        let labelSelector = ``
-        if (clusterName) {
-          labelSelector += `,langop.io/cluster=${clusterName}`
-        }
+        const labelSelector = clusterName ? `langop.io/cluster=${clusterName}` : undefined
 
         watchCleanup = await watchService.watchLanguageTools(
           {
@@ -102,14 +100,16 @@ export async function GET(request: NextRequest) {
               }, 'error')
             }
 
-            // Retry immediately if client is still connected
-            // This prevents missing events during reconnection
+            // Retry only if client is still connected
             if (!request.signal.aborted && isActive()) {
-              console.log('🔄 Reconnecting tool watch immediately...')
-              retryTimeout = setTimeout(startWatch, 100) // 100ms delay to prevent tight loop
+              const delay = Math.min(1000 * Math.pow(2, retryCount), 30000)
+              retryCount++
+              console.log(`🔄 Reconnecting tool watch in ${delay}ms...`)
+              retryTimeout = setTimeout(startWatch, delay)
             }
           }
         )
+        retryCount = 0
       } catch (error) {
         console.error('Failed to start tool watch:', error)
 

